`pdctl/TestScheduler` is unstable

https://github.com/tikv/pd/actions/runs/3368718923/jobs/5587582938

```
--- FAIL: TestScheduler (5.65s)
    scheduler_test.go:410: 
        	Error Trace:	scheduler_test.go:410
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"byte-rate-rank-step-ratio":0.05, "count-rank-step-ratio":0.01, "dst-tolerance-ratio":1.05, "enable-for-tiflash":"true", "forbid-rw-type":"read", "great-dec-ratio":0.95, "key-rate-rank-step-ratio":0.05, "max-peer-number":1000, "max-zombie-rounds":3, "min-hot-byte-rate":100, "min-hot-key-rate":10, "min-hot-query-rate":10, "minor-dec-ratio":0.99, "query-rate-rank-step-ratio":0.05, "rank-formula-version":"v1", "read-priorities":[]interface {}{"key", "byte"}, "src-tolerance-ratio":1.02, "strict-picking-store":"true", "write-leader-priorities":[]interface {}{"key", "byte"}, "write-peer-priorities":[]interface {}{"byte", "key"}}
        	            	actual  : map[string]interface {}{"byte-rate-rank-step-ratio":0.05, "count-rank-step-ratio":0.01, "dst-tolerance-ratio":1.05, "enable-for-tiflash":"true", "forbid-rw-type":"read", "great-dec-ratio":0.95, "key-rate-rank-step-ratio":0.05, "max-peer-number":1000, "max-zombie-rounds":3, "min-hot-byte-rate":100, "min-hot-key-rate":10, "min-hot-query-rate":10, "minor-dec-ratio":0.99, "query-rate-rank-step-ratio":0.05, "rank-formula-version":"v1", "read-priorities":[]interface {}{"key", "byte"}, "src-tolerance-ratio":1.02, "strict-picking-store":"true", "write-leader-priorities":[]interface {}{"query", "byte"}, "write-peer-priorities":[]interface {}{"byte", "key"}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -23,3 +23,3 @@
        	            	  (string) (len=23) "write-leader-priorities": ([]interface {}) (len=2) {
        	            	-  (string) (len=3) "key",
        	            	+  (string) (len=5) "query",
        	            	   (string) (len=4) "byte"
        	Test:       	TestScheduler
FAIL
```

--- a/tests/pdctl/scheduler/scheduler_test.go
+++ b/tests/pdctl/scheduler/scheduler_test.go
@@ -400,6 +400,7 @@ func TestScheduler(t *testing.T) {
 	mustExec([]string{"-u", pdAddr, "scheduler", "config", "balance-hot-region-scheduler", "set", "forbid-rw-type", "read"}, nil)
 	mustExec([]string{"-u", pdAddr, "scheduler", "config", "balance-hot-region-scheduler"}, &conf1)
 	re.Equal(expected1, conf1)
+
 	// test compatibility
 	re.Equal("2.0.0", leaderServer.GetClusterVersion().String())
 	for _, store := range stores {
@@ -408,22 +409,20 @@ func TestScheduler(t *testing.T) {
 		pdctl.MustPutStore(re, leaderServer.GetServer(), store)
 	}
 	re.Equal("5.2.0", leaderServer.GetClusterVersion().String())
-	mustExec([]string{"-u", pdAddr, "scheduler", "config", "balance-hot-region-scheduler"}, &conf1)
 	// After upgrading, we should not use query.
-	expected1["read-priorities"] = []interface{}{"query", "byte"}
-	re.NotEqual(expected1, conf1)
-	expected1["read-priorities"] = []interface{}{"key", "byte"}
-	re.Equal(expected1, conf1)
+	mustExec([]string{"-u", pdAddr, "scheduler", "config", "balance-hot-region-scheduler"}, &conf1)
+	re.Equal(conf1["read-priorities"], []interface{}{"key", "byte"})
 	// cannot set qps as write-peer-priorities
 	echo = mustExec([]string{"-u", pdAddr, "scheduler", "config", "balance-hot-region-scheduler", "set", "write-peer-priorities", "query,byte"}, nil)
 	re.Contains(echo, "query is not allowed to be set in priorities for write-peer-priorities")
 	mustExec([]string{"-u", pdAddr, "scheduler", "config", "balance-hot-region-scheduler"}, &conf1)
-	re.Equal(expected1["write-peer-priorities"], []interface{}{"byte", "key"})
+	re.Equal(conf1["write-peer-priorities"], []interface{}{"byte", "key"})
 
 	// test remove and add
-	mustExec([]string{"-u", pdAddr, "scheduler", "remove", "balance-hot-region-scheduler"}, nil)
-	mustExec([]string{"-u", pdAddr, "scheduler", "add", "balance-hot-region-scheduler"}, nil)
-	re.Equal(expected1, conf1)
+	echo = mustExec([]string{"-u", pdAddr, "scheduler", "remove", "balance-hot-region-scheduler"}, nil)
+	re.Contains(echo, "Success")
+	echo = mustExec([]string{"-u", pdAddr, "scheduler", "add", "balance-hot-region-scheduler"}, nil)
+	re.Contains(echo, "Success")
 
 	// test balance leader config
 	conf = make(map[string]interface{})

